sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -139,28 +132,78 @@ namespace Azure { namespace Security { namespace Attestation {
+
     /** @brief Construct a new Attestation Client object from an existing attestation client.
      *
      * @param attestationClient An existing attestation client.
      */
-    explicit AttestationClient(AttestationClient const& attestationClient)
+    AttestationClient(AttestationClient const& attestationClient)
         : m_endpoint(attestationClient.m_endpoint), m_apiVersion(attestationClient.m_apiVersion),
           m_pipeline(attestationClient.m_pipeline),
           m_tokenValidationOptions(attestationClient.m_tokenValidationOptions){};
 
-    /**
-     * @brief Returns the API version the client was configured with.
+    /** @brief Construct a new Attestation Client object
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential The authentication method to use (required for TPM attestation).
+     * @param options The options to customize the client behavior.
+     */
+    static AttestationClient const Create(
+        std::string const& endpoint,
+        std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+        AttestationClientOptions options = AttestationClientOptions(),
+        Azure::Core::Context const& constext = Azure::Core::Context{});
+
+    /** @brief Construct a new anonymous Attestation Client object
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param options The options to customize the client behavior.
+     *
+     * @note TPM attestation requires an authenticated attestation client.
+     */
+    static AttestationClient const Create(
+        std::string const& endpoint,
+        AttestationClientOptions options = AttestationClientOptions(),
+        Azure::Core::Context const& constext = Azure::Core::Context{});
+
+    /** @brief Construct a new Attestation Client object
      *
-     * @returns The API version used when communicating with the attestation service.
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential The authentication method to use (required for TPM attestation).
+     * @param options The options to customize the client behavior.
      */
-    std::string const& ClientVersion() const { return m_apiVersion; }
+    static AttestationClient* const CreatePointer(
+        std::string const& endpoint,
+        std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+        AttestationClientOptions options = AttestationClientOptions(),
+        Azure::Core::Context const& constext = Azure::Core::Context{});
+
+    /** @brief Construct a new anonymous Attestation Client object
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param options The options to customize the client behavior.
+     *
+     * @note TPM attestation requires an authenticated attestation client.
+     */
+    static AttestationClient* const CreatePointer(
```

Comment:
should it be at least `std::unique_ptr<AttestationClient>` ?  Who's going to delete this?

Reply:
I was concerned about making a choice about what to return to the caller and instead was relying on documentation. However reading C.50, I've changed it to always return a std::unique_ptr - callers that want a shared_ptr can transfer it themselves.